src/appmixer/utils/csv/AddRow/AddRow.js

```diff
@@ -23,34 +23,47 @@ module.exports = {
             parseNumbers,
             parseBooleans
         });
+        let lock;
+        let lockExtendInterval;
+        try {
+            lock = await context.lock(fileId, {
+                ttl: parseInt(context.config.lockTTL, 10) || 1000 * 60 * 1,
+                retryDelay: 1000
+            });
+            await processor.loadHeaders();
 
-        await processor.loadHeaders();
-
-        let rowAsArray;
+            let rowAsArray;
 
-        if (withHeaders) {
-            const headers = processor.getHeaders();
-            const parsed = expressionTransformer(rowWithColumns);
-            rowAsArray = headers.map(item => '');
-            parsed.forEach(item => {
-                const idx = processor.getHeaderIndex(item.column);
-                rowAsArray[idx] = item.value;
-            });
-        } else {
-            rowAsArray = row.split(delimiter);
-        }
+            if (withHeaders) {
+                const headers = processor.getHeaders();
+                const parsed = expressionTransformer(rowWithColumns);
+                rowAsArray = headers.map(item => '');
+                parsed.forEach(item => {
+                    const idx = processor.getHeaderIndex(item.column);
+                    rowAsArray[idx] = item.value;
+                });
+            } else {
+                rowAsArray = row.split(delimiter);
+            }
 
-        for (let i = 0; i < rowAsArray.length; i++) {
-            const item = rowAsArray[i];
-            if (item === undefined || item === null) {
-                rowAsArray[i] = '';
+            for (let i = 0; i < rowAsArray.length; i++) {
+                const item = rowAsArray[i];
+                if (item === undefined || item === null) {
+                    rowAsArray[i] = '';
+                }
             }
+            lockExtendInterval = setInterval(async () => {
+                if (lock) {
+                    await lock.extend(parseInt(context.config.lockExtendTime, 10) || 1000 * 60 * 1);
+                }
+            }, parseInt(context.config.lockExtendInterval, 10) || 30000);
```

Comment:
The default values are fine, interval is half the TTL. But If I set `lockExtendInterval` in the backoffice, then the `interval` is the same as the TTL, which is not correct. Could lead to situation, when you open the lock for a while.